pandas/core/arrays/categorical.py

```diff
@@ -523,6 +523,8 @@ def astype(self, dtype: Dtype, copy: bool = True) -> ArrayLike:
         if is_integer_dtype(dtype) and self.isna().any():
             msg = "Cannot convert float NaN to integer"
             raise ValueError(msg)
+        if is_extension_array_dtype(dtype):
```

Comment:
Can you move this above the previous `if` statement.  With the current order we can't astype to the new nullable integer dtype, which we should be able to do, e.g.
```python
In [2]: cat = pd.Categorical([1, 2, np.nan])

In [3]: cat.astype('Int64')
---------------------------------------------------------------------------
ValueError: Cannot convert float NaN to integer
```

Reply:
Done, should we add a test for this kind of behavior?